src/audio/drc/drc.c

```diff
@@ -138,6 +139,7 @@ static int drc_setup(struct drc_comp_data *cd, uint16_t channels, uint32_t rate)
  * End of DRC setup code. Next the standard component methods.
  */
 
+__sof_llext_dram_text(CONFIG_COMP_DRC_MODULE)
```

Comment:
Btw, - this should be something small and simple that should complie in/out based on Kconfig in the definition. e.g.
```__cold static int drc_init(struct processing_module *mod)```

Reply:
@lgirdwood the name can be changed of course, and sure, it resolves to nothing with suitable Kconfig, but it has to depend on the respective Kconfig option. Effectively what it does is
```
#if CONFIG_COMP_X_MODULE
__section(".text.dram")
#endif
```
so in each module it needs to know which Kconfig option to depend upon. I also made it explicit, that it's a SOF macro, it only works with LLEXT, and this one is for functions, not for data. The easiest would be to have two different macros for functions and for data. And "cold" sounds like it can be applied to either. I'm not sure we can resolve automatically at build time data from code.

Reply:
It needs to be short and sweet, keep the CONFIG part in the header. It should be one word. e.g. Linux uses __init for initialization sections.

Reply:
> It needs to be short and sweet, keep the CONFIG part in the header. It should be one word. e.g. Linux uses __init for initialization sections.

@lgirdwood but you'd need such a macro for each component. And some components might not even have their own headers. So are you proposing to put something like
```
#if CONFIG_COMP_X_MODULE
#define __cold __section("...")
#else
#define __cold
#endif
```
in each component and eventually in each LLEXT-enabled source file?

Reply:
This one-line version is very nice!

Reply:
We shouldn't need per module definitions, we should be able to ```#include <llext_module.h``` (or whatever) and in that header define e.g.
```c
#if defined (CC_SAYS_I_AM_BUILDING_A_MODULE)
#define __cold __section("...")
#else
#define __cold
#endif
```